src/core/datatypes/math/OrderedEventTimes.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+ * EventTime.cpp
+ *
+ *  Created on: Apr 9, 2020
+ *      Author: mike
+ */
+
+#include <ostream>
+
+#include "OrderedEventTimes.h"
+#include "TypedDagNode.h"
+
+namespace RevBayesCore {
+
+OrderedEventTimes::OrderedEventTimes()
+{
+}
+
+OrderedEventTimes::~OrderedEventTimes()
+{
+}
+
+OrderedEventTimes* OrderedEventTimes::clone(void) const
+{
+    return new OrderedEventTimes( *this );
+}
+
+bool OrderedEventTimes::operator==(const OrderedEventTimes &oet) const
+{
+	return event_times == oet.event_times;
+}
+
+bool OrderedEventTimes::operator!=(const OrderedEventTimes &oet) const
+{
+	return event_times != oet.event_times;
+}
+
+bool OrderedEventTimes::operator<(const  OrderedEventTimes &oet) const
+{
+	return false;
+}
+
+
+bool OrderedEventTimes::addEvent(double time)
+{
+	// check if the event already exists
+	std::set<double>::iterator it = event_times.find(time);
```

Comment:
Comparing double precision numbers for exact equality makes me a bit nervous.  Does this seem to work?

Its possible that this is OK nowadays, but it used to cause trouble.  The Intel FPU uses 80-bit precision in on-chip registers, but rounds them down to 64-bit precision when storing them to memory.  This could mean that numbers that were identical at 64-bit precision would compare not-equal at 80-bit precision.

Most floating-point code these days ignores the FPU and uses MMX/SSE instead, which doesn't have 80-bit precision.  But my impression was that people generally avoid comparing double's for exact equality...

Reply:
Seconded. Would comparing the numbers up to `DBL_EPSILON` or some similarly selected delta be a good solution?